apps/server/src/services/ElasticSearchClient.ts

```diff
@@ -278,3 +276,45 @@ export async function getRecommendedInstitutions(
 
   return institutions
 }
+
+export async function deleteRemovedInstitutions(
+  newInstitutions: CachedInstitution[]
+) {
+  const newInsIds = newInstitutions.map((ins: CachedInstitution) => ins.ucp_id)
+  const oldInstitutions = JSON.parse(
+    readFileSync(getInstitutionFilePath(), 'utf8')
+  )
+  const oldInsIds = oldInstitutions.map((ins: CachedInstitution) => ins.ucp_id)
+  const deletedInstitutionsIds = oldInsIds.filter(
```

Comment:
I believe this is a O(n^2) solution, and we're looking at 13,000+ institutions in the array. Worst case scenario this performs 169 million checks.

I'd use array .reduce to create an object with keys as the id and true as the value for everything that's included in newInsIds. You can use that object in this filter function instead of .includes and that brings us down to O(n)

Reply:
You're right that is terribly inefficient. I did a little research and `Set.has` is an O(1) operation so I can just convert the array to a Set and basically do the same thing but it feels a little simpler to me than using reduce although I think it accomplishes the same thing